app/models/profile.rb

```diff
@@ -40,7 +40,7 @@ def activate
     is_reproof = Profile.find_by(user_id: user_id, active: true)
     transaction do
       Profile.where(user_id: user_id).update_all(active: false)
-      update!(
+      update(
```

Comment:
Yep, I suspect it is intentional and we were trying to avoid a case where the first update succeeds and the second one fails, leaving the database with multiple active profiles.

Reply:
should I propagate it then on callers?

Reply:
I've read the linked context in the original comment and I'm still not entirely clear on the motivation for these changes.

But yeah, I'd agree that if tests are passing as-is despite how we may want the transaction to work, sounds like the test coverage might be lacking.

Reply:
some prior art:
https://github.com/18F/identity-idp/blob/35de604ef0e8aee4e5991ba90d53d98c6ee8463d/app/jobs/reports/base_report.rb#L7-L10